cluster/map_storage.go

```diff
@@ -222,11 +222,16 @@ func (s *MapStorage) RetrieveNodesByMetadata(metadata map[string]string) ([]Node
 	defer s.nMut.Unlock()
 	filteredNodes := []Node{}
 	for _, node := range s.nodes {
-		for key, value := range metadata {
-			nodeVal, ok := node.Metadata[key]
-			if ok && nodeVal == value {
-				filteredNodes = append(filteredNodes, node)
+		if func(nodeMetadata map[string]string) bool {
```

Comment:
I think declaring a function inside an if statement is a little bit too clever. What do you think about declaring a top level function `func hasAllMetadata(base, wanted map[string]string) bool` instead?